Golang: no tests to run

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Every test I run using the "play" button returns a "[no tests to run]" message in the terminal.

`main.go`
```go
package main

func main() {
}

func TestBool(b bool) bool {
	return b
}
```

`main_test.go`
```go
package main

import "testing"

func TestThis(t *testing.T) {
	if !TestBool(false) {
		println("TestBool(true) failed")
		t.Fail()
	}
}
```

This is the output when I click the play button:

```
ok  	github.com/gerep/go-test	0.002s [no tests to run]

⏵ Task `go test . -run TestThis` finished successfully
⏵ Command: '/bin/bash -i -c `go test . -run '^TestThis$'`'
```

This is the output when I run the command `go test . -run '^TestThis$'`:

```
go test . -run TestThis
TestBool(true) failed
--- FAIL: TestThis (0.00s)
FAIL
FAIL	github.com/gerep/go-test	0.002s
FAIL
```


### Environment

Zed: v0.144.3 (Zed)
OS: Linux X11 pop 22.04
Memory: 30.9 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->

<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

Found the culprit: https://github.com/zed-industries/zed/pull/14055

I think with that change, instead of executing `go test . -run '^TestThis$'` what ends up being executes is equivalent to this: `go test . -run \'^TestThis$\'`.

```
$ go test . -run '^TestThis$'
--- FAIL: TestThis (0.00s)
    main_test.go:6: lol
FAIL
FAIL    testinggotests  0.165s
FAIL
$ go test . -run \'^TestThis$\'
ok      testinggotests  0.178s [no tests to run]
```

That's zsh.

But I think there's a possible fix:

fish:

```fish
# doesn't work:
$ go test . -run ^TestThis$
fish: Expected a variable name after this $.
go test . -run ^TestThis$
                        ^
# works:                        
$ go test . -run ^TestThis\$
--- FAIL: TestThis (0.00s)
    main_test.go:6: lol
FAIL
FAIL    testinggotests  0.176s
FAIL

# doesn't work:
$ go test . -run '^TestThis\$'
ok      testinggotests  0.175s [no tests to run]

# works:
$ go test . -run '^TestThis$'
--- FAIL: TestThis (0.00s)
    main_test.go:6: lol
FAIL
FAIL    testinggotests  0.175s
FAIL
```

And `\$` also works on ZSH:

```
~/tmp/testing-go-tests φ go test . -run ^TestThis$
--- FAIL: TestThis (0.00s)
    main_test.go:6: lol
FAIL
FAIL    testinggotests  0.176s
FAIL
~/tmp/testing-go-tests φ go test . -run ^TestThis\$
--- FAIL: TestThis (0.00s)
    main_test.go:6: lol
FAIL
FAIL    testinggotests  0.179s
FAIL
```

--- a/crates/languages/src/go.rs
+++ b/crates/languages/src/go.rs
@@ -518,7 +518,7 @@ impl ContextProvider for GoContextProvider {
                     "test".into(),
                     GO_PACKAGE_TASK_VARIABLE.template_value(),
                     "-run".into(),
-                    format!("'^{}$'", VariableName::Symbol.template_value(),),
+                    format!("^{}\\$", VariableName::Symbol.template_value(),),
                 ],
                 tags: vec!["go-test".to_owned()],
                 ..TaskTemplate::default()
@@ -549,7 +549,7 @@ impl ContextProvider for GoContextProvider {
                     "-v".into(),
                     "-run".into(),
                     format!(
-                        "'^{}$/^{}$'",
+                        "^{}\\$/^{}\\$",
                         VariableName::Symbol.template_value(),
                         GO_SUBTEST_NAME_TASK_VARIABLE.template_value(),
                     ),
@@ -570,7 +570,7 @@ impl ContextProvider for GoContextProvider {
                     "-benchmem".into(),
                     "-run=^$".into(),
                     "-bench".into(),
-                    format!("'^{}$'", VariableName::Symbol.template_value()),
+                    format!("^{}\\$", VariableName::Symbol.template_value()),
                 ],
                 tags: vec!["go-benchmark".to_owned()],
                 ..TaskTemplate::default()

